src/fiber/channel.rs

```diff
@@ -2683,6 +2730,34 @@ impl PendingTlcs {
             .cloned()
             .collect()
     }
+
+    pub fn get_oldest_failed_tlcs(&self) -> Vec<TLCId> {
+        let mut failed_tlcs = self
+            .tlcs
+            .iter()
+            .filter(|tlc| {
```

Comment:
can we use `filter_map` to return the tlc which is `is_fail_remove_confirmed` only?